File: recursion/finding-substrings/python/StringSubsequence.py

def get_subsequence(word):

    if len(word) == 0:
        return ''

    first = word[0]
    rest = get_subsequence(word[1:])

    result = ''

    for strsub in rest.split(','):
        result += ',' + first + strsub
        result += ',' + strsub

    result = result[1:]

    return result

print(get_subsequence("XYZ"))

File: recursion/finding-substrings/javascript/StringSubsequence.js
function getSubsequence(word){

    if(word == '')
    {
        return '';
    }

    const first = word.charAt(0);

    const rest = getSubsequence(word.substring(1));

    let result = '';

    rest.split(',').forEach(subseq => {

        result += ','+ first + subseq;
        result += ','+ subseq; 

    });

    return result.substring(1);
}

subsequences = getSubsequence('abc');
console.log(subsequences);
